dspace-api/src/main/java/org/dspace/util/SolrImportExport.java

```diff
@@ -0,0 +1,454 @@
+				                 "The absolute path for the directory to use for import or export. If none is given, [dspace]/solr-export is used.");
+
+		try
+		{
+			CommandLine line = parser.parse(options, args);
+			if (line.hasOption(HELP_OPTION))
+			{
+				printHelpAndExit(options, 0);
+			}
+
+			if (!line.hasOption(INDEX_NAME_OPTION))
+			{
+				System.err.println("This command requires the index-name option but none was present.");
+				printHelpAndExit(options, 1);
+			}
+			String[] indexNames = line.getOptionValues(INDEX_NAME_OPTION);
+
+			String action = line.getOptionValue(ACTION_OPTION, "export");
+			if ("import".equals(action))
+			{
+				for (String indexName : indexNames)
+				{
+					try
+					{
+						String fromDir = makeDirectoryName(line.getOptionValue(DIRECTORY_OPTION));
+						String solrUrl = makeSolrUrl(indexName);
+						importIndex(indexName, fromDir, line.hasOption(CLEAR_OPTION), solrUrl);
+					}
+					catch (IOException | SolrServerException | SolrImportExportException e)
+					{
+						System.err.println("Problem encountered while trying to import index " + indexName + ".");
+						e.printStackTrace(System.err);
+					}
+				}
+			}
+			else if ("export".equals(action))
+			{
+				for (String indexName : indexNames)
+				{
+					try
+					{
+						String toDir = makeDirectoryName(line.getOptionValue(DIRECTORY_OPTION));
+						String solrUrl = makeSolrUrl(indexName);
+						String timeField = makeTimeField(indexName);
+						exportIndex(indexName, toDir, solrUrl, timeField);
+					}
+					catch (SolrServerException | IOException | SolrImportExportException e)
+					{
+						System.err.println("Problem encountered while trying to export index " + indexName + ".");
+						e.printStackTrace(System.err);
+					}
+				}
+			}
+			else if ("reindex".equals(action))
+			{
+				for (String indexName : indexNames)
+				{
+					String tempIndexName = indexName + "-temp";
+
+					String origSolrUrl = makeSolrUrl(indexName);
+					String baseSolrUrl = StringUtils.substringBeforeLast(origSolrUrl, "/"); // need to get non-core solr URL
+					String tempSolrUrl = baseSolrUrl + "/" + tempIndexName;
+
+					String solrInstanceDir = ConfigurationManager.getProperty("dspace.dir") + File.separator + "solr" + File.separator + indexName;
+
+					String exportDir = makeDirectoryName(line.getOptionValue(DIRECTORY_OPTION)); // TODO optionally, keep the export?
+					String timeField = makeTimeField(indexName);
+
+					// Create a temp directory to store temporary core data
+					File tempDataDir = new File(ConfigurationManager.getProperty("dspace.dir") + File.separator + "temp" + File.separator + "solr-data");
+					tempDataDir.mkdirs();
+
+					// create a temporary core to hold documents coming in during the reindex
+					try {
+						HttpSolrServer adminSolr = new HttpSolrServer(baseSolrUrl);
+						CoreAdminRequest.Create createRequest = new CoreAdminRequest.Create();
+						createRequest.setInstanceDir(solrInstanceDir);
```

Comment:
Hi Andrea, it looks like this core swapping functionality assumes that local Solr is used (and I don't see how it would work efficiently otherwise). This might be the case for most users, but it won't work if you're using a remote Solr instance. There should be a warning for such case. One heuristics for detecting whether local or remote Solr is used could be to see if there's localhost in solr url - not perfect by far, but anything fancy I can think of involves comparing local files to Solr responses which is complex and likely to break.